### PR TITLE
Adds remaining SWIG wrappers

### PR DESCRIPTION
**Pull request by jessemapel**
This just provides basic wrappers. There are still a handful of things that need to be added including the C++ JSON <--> Python Dict conversions.

### CODE REVIEW

File: .github/workflows/ctests.yml [hunk 1]
```diff
@@ -76,6 +76,7 @@ jobs:
             test -e lib/libSpiceQL.so
           fi
           test -e include/SpiceQL/sugar_spice.h
+          rm -rf ${{github.workspace}}/build
```
Comment (jessemapel) [inline .github/workflows/ctests.yml:79]: I saw that the wrappers were still pointing back to the build version of the library and not the install version, so this should help to catch stuff like that.

**Comment by jessemapel** [comment]
More #56 work. Also helpful for #55

File: bindings/python/query.i [hunk 2]
```diff
@@ -0,0 +1,7 @@
+%module(package="pyspiceql") query
+
+%{
+  #include "query.h"
+%}
+
+%include "query.h"
```
Comment (Kelvinrr) [inline bindings/python/query.i:7]: Not super familiar with SWIG, but is this all the is required to wrap functions in a header? Or do you expect the type conversion to be split in different PRs?

Comment (jessemapel) [reply bindings/python/query.i:7]: This handles all of the basic conversions like vector to list and string to Python strings. The more complex conversions I'll do either in this PR or a third if it's too complex.

File: SpiceQL/include/io.h [hunk 3]
```diff
@@ -48,12 +48,6 @@ namespace SpiceQL {
                  std::optional<std::vector<std::vector<double>>> stateVelocities,
                  std::optional<std::string> segmentComment);
 
-      /**
-       * Combine the state positions and velocities into a single vector
-       *
-       * @return Single vector with {X1, Y1, Z1, dX1, dY1, dZ1, X2, Y2, Z2, dX2, dY2, dZ2, ...}
-       */
-      static std::vector<std::vector<double>> concatStates (std::vector<std::vector<double>> statePositions, std::vector<std::vector<double>> stateVelocities);
```
Comment (jessemapel) [inline SpiceQL/include/io.h:56]: This was in the header as a static method on the class but it wasn't actually implemented and was causing undefined symbol errors in the swig wrapper.

The function in the source file was actually just a free function so I went with that.

File: SpiceQL/include/spice_types.h [hunk 4]
```diff
@@ -52,12 +52,12 @@ namespace SpiceQL {
       /**
        * used for converting string and int kernal quality
        */
-      const static std::vector<std::string> QUALITIES;
+      static const std::vector<std::string> QUALITIES;
```
Comment (jessemapel) [inline SpiceQL/include/spice_types.h:55]: This is totally fine C++ but not C, so SWIG doesn't like it. Hence the swap. This won't impact the library.

File: bindings/python/CMakeLists.txt [hunk 5]
```diff
@@ -26,16 +31,6 @@ else()
...
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/__init__.py.in ${CMAKE_CURRENT_BINARY_DIR}/__init__.py)
-
-# Custom Python installation target
-add_custom_target(install-python
-        DEPENDS pyspiceql
-        COMMAND python ${SETUP_PY_OUT} install)
\ No newline at end of file
+# Install the Python wrapper
+install(TARGETS pyspiceql DESTINATION ${Python3_SITEARCH})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/pyspiceql.py DESTINATION ${Python3_SITEARCH})
```
Comment (jessemapel) [inline bindings/python/CMakeLists.txt:36]: The `Python_SITEARCH` variable is set in findPython3. So, this will install into whatever Python it links against, not the one specified in the install directory. I think that's okay but there could be strange issues with conda-build. For now it works when I build and install into a local environment so I think we're okay.

**Comment by Kelvinrr** [comment]
I would just test what you think will be a problem. I'm still not 100% sure what a good way to approach testing is on the long run but I imagine testing this through the rest api tests in the future will probably be best for now?

**Review by Kelvinrr — APPROVED** [review]
